core/lib/state/src/rocksdb/mod.rs

```diff
@@ -55,22 +60,85 @@ impl NamedColumnFamily for StateKeeperColumnFamily {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+struct StateValue {
+    pub value: H256,
+    pub enum_index: Option<u64>,
+}
+
+impl StateValue {
+    pub fn deserialize(bytes: &[u8]) -> Self {
+        if bytes.len() == 32 {
+            Self {
+                value: H256::from_slice(bytes),
+                enum_index: None,
+            }
+        } else {
+            Self {
+                value: H256::from_slice(&bytes[..32]),
+                enum_index: Some(u64::from_be_bytes(bytes[32..40].try_into().unwrap())),
+            }
+        }
+    }
+
+    pub fn serialize(&self) -> Vec<u8> {
+        self.value
```

Comment:
Bikeshedding: IMO, this is less readable than an approach with `extend_from_slice`, but that's just a personal opinion.

```rust
let mut buffer = Vec::with_capacity(40);
buffer.extend_from_slice(self.value.as_bytes());
if let Some(index) = self.enum_index {
    buffer.extend_from_slice(&index.to_be_bytes());
}
buffer
```

Reply:
Agree, it's more readable indeed